File: app/src/main/java/be/hcpl/android/filmtag/util/TextUtil.java
package be.hcpl.android.filmtag.util;

import java.text.DecimalFormat;

public class TextUtil {

    private static DecimalFormat frameFormat = new DecimalFormat("00");
    private static DecimalFormat apertureFormat = new DecimalFormat("0.#");

    public static String formatFrameNumber(int frameNumber) {
        return "#" + frameFormat.format(frameNumber);
    }

    public static String formatAperture(double aperture) {
        return "f/" + apertureFormat.format(aperture);
    }

    public static String formatShutter(int shutter, boolean longExposure) {
        return (longExposure ? "" : "1/") + shutter + " s";
    }
}

File: app/src/main/java/be/hcpl/android/filmtag/util/TextUtil.kt
package be.hcpl.android.filmtag.util

import java.text.DecimalFormat

object TextUtil {

    // TODO convert to some sort of transformer and inject

    private val frameFormat = DecimalFormat("00")
    private val apertureFormat = DecimalFormat("0.#")

    fun formatFrameNumber(frameNumber: Int): String {
        return "#" + frameFormat.format(frameNumber.toLong())
    }

    fun formatAperture(aperture: Double): String {
        return "f/" + apertureFormat.format(aperture)
    }

    fun formatShutter(shutter: Int, longExposure: Boolean): String {
        return (if (longExposure) "" else "1/") + shutter + " s"
    }
}
